internal/stanza/converter.go

```diff
@@ -246,56 +229,44 @@ func (c *Converter) workerLoop() {
 func (c *Converter) batchLoop() {
 	defer c.wg.Done()
 
-	ticker := time.NewTicker(c.flushInterval)
-	defer ticker.Stop()
-
 	for {
 		select {
-		case wi, ok := <-c.batchChan:
+		case workerItems, ok := <-c.batchChan:
 			if !ok {
 				return
 			}
 
-			pLogs, ok := c.data[wi.ResourceID]
-			if ok {
-				lr := pLogs.ResourceLogs().
-					At(0).InstrumentationLibraryLogs().
-					At(0).Logs().AppendEmpty()
-				wi.LogRecord.CopyTo(lr)
-			} else {
-				pLogs = pdata.NewLogs()
-				logs := pLogs.ResourceLogs()
-				rls := logs.AppendEmpty()
-
-				resource := rls.Resource()
-				resourceAtts := resource.Attributes()
-				resourceAtts.EnsureCapacity(len(wi.Resource))
-				for k, v := range wi.Resource {
-					resourceAtts.InsertString(k, v)
+			for _, wi := range workerItems {
+				pLogs, ok := c.data[wi.ResourceID]
```

Comment:
Is this safe? I realize it was already like this, but it looking at it now it seems unsafe. If it is safe, can we have a comment explaining why / what assumptions must be followed in future changes in order to keep it safe?

Reply:
Can you expand on what you think might be unsafe here?

Reply:
Map access w/o a mux.

Reply:
Yeah, I had the same thought when I looked at this first. The aggregation goroutine only has one instance of it running; The aggregation really could be run in parallel and can actually just be folded into the worker loop, which would make this less concerning at a glance.

I think instead of adding a comment, I'll just remove c.data and move it to be local to the aggregation loop, so it's obvious there isn't any concurrent access.

Reply:
Made the resource aggregation map local to ensure safety and improve readability.